Type 'IgxPivotGridComponent' is not assignable to type 'PivotGridType'

## Description  
Open the attached sample


## Steps to reproduce  

1. npm i
2. npm run build

## Result  

```console
X [ERROR] TS2322: Type 'IgxPivotGridComponent' is not assignable to type 'PivotGridType'.
  Types of property 'columnList' are incompatible.
    Type 'QueryList<IgxColumnComponent>' is not assignable to type 'QueryList<ColumnType>'.
      Type 'IgxColumnComponent' is not assignable to type 'ColumnType'.
        The types of 'grid.data' are incompatible between these types.
          Type 'any[] | null' is not assignable to type 'any[] | undefined'.
            Type 'null' is not assignable to type 'any[] | undefined'. [plugin angular-compiler]

    src/app/pivot-grid/pivot-grid.component.html:3:28:
      3 │   <igx-pivot-data-selector [grid]="pivotGrid"></igx-pivot-data-sele...
```

## Expected result  
Running sample

## Attachments 
[AllComponents_PivotGrid.zip](https://github.com/user-attachments/files/23821524/AllComponents_PivotGrid.zip)

--- a/projects/igniteui-angular/core/src/data-operations/grid-types.ts
+++ b/projects/igniteui-angular/core/src/data-operations/grid-types.ts
@@ -93,7 +93,7 @@ export enum ColumnPinningPosition {
 export interface GridTypeBase {
     primaryKey?: string;
     id?: string;
-    data?: any[];
+    data: any[] | null;
     [key: string]: any;
 }
 

